x-pack/plugins/cases/public/components/create/form_context.test.tsx

```diff
@@ -612,11 +620,32 @@ describe('Create case', () => {
         onChoicesSuccess(useGetChoicesResponse.choices);
       });
 
+      await waitFor(() => {
+        expect(
+          within(screen.getByTestId('severitySelect')).getByRole('option', { name: '2 - High' })
+        );
+
+        expect(
+          within(screen.getByTestId('urgencySelect')).getByRole('option', { name: '2 - High' })
+        );
+
+        expect(
+          within(screen.getByTestId('impactSelect')).getByRole('option', { name: '2 - High' })
+        );
+
+        expect(screen.getByRole('option', { name: 'Software' }));
+      });
+
       ['severitySelect', 'urgencySelect', 'impactSelect'].forEach((subj) => {
         userEvent.selectOptions(screen.getByTestId(subj), ['2']);
       });
 
       userEvent.selectOptions(screen.getByTestId('categorySelect'), ['software']);
+
+      await waitFor(() => {
+        expect(screen.getByRole('option', { name: 'Operation System' }));
```

Comment:
_nit_ I would rather have `within(...)` being used everywhere that we "wait for" a `.getByRole(...)`

So this whole block could alternatively be:

```
const subcategorySelect = screen.getByTestId('subcategorySelect');

await waitFor(() => {
  expect(within(subcategorySelect).getByRole('option', { name: 'Operation System' }));
});

userEvent.selectOptions(subcategorySelect, ['os']);
```

Reply:
Fixed!